lib/app/endpoint/home.rb

```diff
@@ -23,33 +24,33 @@ class Home < Endpoint
         # This must be early so it doesn't get picked up by the other routes
         get '/oauth2/:key/:endpoint/?' do
           instance = nil
-          if params[:endpoint] == 'redirect' && !params[:state].nil?
+          error_message = nil
+
+          if params[:endpoint] == 'redirect'
             instance = Inferno::Models::TestingInstance.first(state: params[:state])
             if instance.nil?
-              error_message = "<p>Inferno has detected an issue with the SMART launch. No actively running launch sequences found with a state of #{params[:state]}. " \
-                              'The authorization server is not returning the correct state variable and therefore Inferno cannot identify which server is currently under test. ' \
-                              "Please click your browser's \"Back\" button to return to Inferno, and click \"Refresh\" to ensure that the most recent test results are visible.</p>"
-
-              error_message += "<p>Error returned by server: <strong>#{params[:error]}</strong>.</p>" unless params[:error].nil?
-
-              error_message += "<p>Error description returned by server: <strong>#{params[:error_description]}</strong>.</p>" unless params[:error_description].nil?
-
-              halt 500, error_message
+              instance = Inferno::Models::TestingInstance.get(cookies[:instance_id_test_set]&.split('/')&.first)
+              if instance.nil?
+                error_message = '<p>Inferno has detected an issue with the SMART launch. ' \
+                                "No actively running launch sequences found with a state of #{params[:state]}. " \
+                                'The authorization server is not returning the correct state variable and ' \
+                                'therefore Inferno cannot identify which server is currently under test. ' \
+                                "Please click your browser's \"Back\" button to return to Inferno, " \
+                                "and click \"Refresh\" to ensure that the most recent test results are visible.</p>" # rubocop: disable Style/StringLiterals
```

Comment:
The rubocop disable only affects this line, right? This line could use single quotes and unescape the double quotes rather than ignoring the cop.

Reply:
Yeah we shouldn't disable the cop unless there is a good reason. Any objection to me moving this to `%()` @Jammjammjamm ?

Reply:
Sounds good to me.